packages/core/upload/server/services/image-manipulation.js

```diff
@@ -187,16 +224,22 @@ const isSupportedImage = (...args) => {
 /**
  *  Applies a simple image transformation to see if the image is faulty/corrupted.
  */
-const isFaultyImage = (file) =>
-  new Promise((resolve) => {
-    file
-      .getStream()
-      .pipe(sharp().rotate())
-      .on('error', () => resolve(true))
-      .pipe(writableDiscardStream())
-      .on('error', () => resolve(true))
-      .on('close', () => resolve(false));
-  });
+const isFaultyImage = async (file) => {
+  if (!file.filepath) {
+    return new Promise((resolve, reject) => {
+      const pipeline = sharp();
+      pipeline.stats().then(resolve).catch(reject);
```

Comment:
If we drop the transform and just do `stats`, we lose a bit of robustness, because there could still be encoding issues we don't catch. At the same time, it's going to be significantly faster and probably good enough for all but a very tiny portion of malformed images.

Reply:
based on the maintainer reco we should be on the safe side 👍  https://github.com/lovell/sharp/issues/1298